File: blazerod/render/src/main/java/top/fifthlight/blazerod/extension/RenderPipelineExt.java
package top.fifthlight.blazerod.extension;

import org.jetbrains.annotations.NotNull;
import top.fifthlight.blazerod.model.resource.VertexType;

import java.util.Optional;
import java.util.Set;

public interface RenderPipelineExt {
    @NotNull
    Optional<VertexType> blazerod$getVertexType();

    @NotNull
    Set<String> blazerod$getStorageBuffers();
}

File: blazerod/render/src/main/kotlin/top/fifthlight/blazerod/extension/RenderPipelineExt.kt
package top.fifthlight.blazerod.extension

import com.mojang.blaze3d.pipeline.RenderPipeline
import top.fifthlight.blazerod.extension.RenderPipelineExt

val RenderPipeline.vertexType
    get() = (this as RenderPipelineExt).`blazerod$getVertexType`()
